packages/@vue/cli-plugin-e2e-nightwatch/__tests__/nightwatchPlugin.spec.js

```diff
@@ -1,13 +1,100 @@
-jest.setTimeout(40000)
+jest.setTimeout(process.env.APPVEYOR ? 300000 : 120000)
 
+const fs = require('fs-extra')
+const path = require('path')
 const create = require('@vue/cli-test-utils/createTestProject')
 
-test('should work', async () => {
-  const project = await create('e2e-nightwatch', {
-    plugins: {
-      '@vue/cli-plugin-babel': {},
-      '@vue/cli-plugin-e2e-nightwatch': {}
+describe('nightwatch e2e plugin', () => {
+  let project
+
+  beforeAll(async () => {
+    project = await create('e2e-nightwatch', {
+      plugins: {
+        '@vue/cli-plugin-babel': {},
+        '@vue/cli-plugin-e2e-nightwatch': {}
+      }
+    })
+
+    await fs.copy(path.join(__dirname, './lib/globals-generated.js'),
+      path.join(project.dir, 'tests/e2e/globals-generated.js'))
+
+    const config = {
+      globals_path: './tests/e2e/globals-generated.js'
     }
+    await project.write('nightwatch.json', JSON.stringify(config))
+  })
+
+  test('should run all tests successfully', async () => {
+    await project.run(`vue-cli-service test:e2e --headless`)
+    let results = await project.read('test_results.json')
+    results = JSON.parse(results)
+    expect(Object.keys(results.modules)).toEqual([
+      'test-with-pageobjects',
+      'test'
+    ])
+  })
+
+  test('should run single test with custom nightwatch.json', async () => {
+    await project.run(`vue-cli-service test:e2e --headless -t tests/e2e/specs/test.js`)
+    let results = await project.read('test_results.json')
+    results = JSON.parse(results)
+    expect(Object.keys(results.modules)).toEqual([
+      'test'
+    ])
+  })
+
+  test('should run single test with custom nightwatch.json and selenium server', async () => {
+    await project.run(`vue-cli-service test:e2e --headless --with-selenium -t tests/e2e/specs/test.js`)
+    let results = await project.read('test_results.json')
+    results = JSON.parse(results)
+
+    let testSettings = await project.read('test_settings.json')
+    testSettings = JSON.parse(testSettings)
+
+    expect(testSettings).toHaveProperty('selenium')
+    expect(testSettings.selenium.start_process).toStrictEqual(true)
+    expect(testSettings.selenium).toHaveProperty('cli_args')
+    expect(Object.keys(results.modules)).toEqual([
+      'test'
+    ])
+  })
+
+  test('should run tests in parallel', async () => {
+    await project.run(`vue-cli-service test:e2e --headless --parallel`)
+    let results = await project.read('test_results.json')
+    results = JSON.parse(results)
+
+    let testSettings = await project.read('test_settings.json')
+    testSettings = JSON.parse(testSettings)
+
+    expect(testSettings.parallel_mode).toStrictEqual(true)
+    expect(testSettings.test_workers).toStrictEqual(true)
+
+    expect(Object.keys(results.modules)).toEqual([
```

Comment:
Now that the tests are run in parallel, the order of `results.modules` is nondeterministic, which could cause test failures.
```suggestion
    expect(Object.keys(results.modules.sort())).toEqual([
```